src/Tablebot/Plugin/Permission.hs

```diff
@@ -0,0 +1,24 @@
+-- |
+-- Module      : Tablebot.Plugin.Permission
+-- Description : A simple interface to allow plugins to handle permissions themselves
+-- Copyright   : (c) Anna Bruce 2021
+-- License     : MIT
+-- Maintainer  : finnjkeating@gmail.com
+-- Stability   : experimental
+-- Portability : POSIX
+--
+-- This contains a simple interface for plugin authors to require a specific level of privilege.
+module Tablebot.Plugin.Permission where
+
+import Discord.Internal.Rest (Message)
+import Tablebot.Handler.Permission
+import Tablebot.Plugin.Discord (sendMessageVoid)
+import Tablebot.Plugin.Types
+
+-- | @requirePermission@ only runs the inputted effect if permissions are matched. Otherwise it returns an error.
+requirePermission :: RequiredPermission -> Message -> DatabaseDiscord () -> DatabaseDiscord ()
+requirePermission perm m a = do
+  p <- getSenderPermission m
+  if userHasPermission perm p
+    then a
+    else sendMessageVoid m "Sorry, you don't have permission to do that."
```

Comment:
Probably best to do this with an Exception when Amelie's exception code is merged in

Reply:
I totally agree, but I wrote this before Amelie had even started...